Core_BetterSquirt/CharaSquirt.cs

```diff
@@ -33,7 +33,13 @@ internal float TouchCooldown
 		}
 
 		private static readonly MethodInfo hitReactionPlayInfo = AccessTools.Method(
-			Type.GetType("VRHandCtrl, Assembly-CSharp") ?? typeof(HandCtrl), "HitReactionPlay", new Type[] { typeof(int), typeof(bool) });
+			Type.GetType("VRHandCtrl, Assembly-CSharp") ?? typeof(HandCtrl), 
+            nameof(HandCtrl.HitReactionPlay), 
+#if KK
+            new Type[] { typeof(int), typeof(bool) });
+#else
+            new Type[] { typeof(AibuColliderKind), typeof(bool) });
```

Comment:
This is necessary for KKS since the method signature changed.

Reply:
I just checked the decompiled code of KKS. 
The signature for both KK and KKS is the same `HitReactionPlay(AibuColliderKind, bool)`, where `AibuColliderKind` is an enum defined in either `HandCtrl` or `VRHandCtrl`.

The reason I use `int` here is so that it would work with both `HandCtrl.AibuColliderKind` and `VRHandCtrl.AibuColliderKind`. This works fine in KK.
Does using `int` not work in KKS?

Reply:
Using int as a substitute for an enum with underlying int for GetMethod and other reflection only works in .Net 3.5 I believe. It's not recommended either way, the proper type should be used instead.

Reply:
It can be done instead with
```cs
		private static readonly MethodInfo hitReactionPlayInfo = 
			Type.GetType("VRHandCtrl, Assembly-CSharp")?.GetMethod(nameof(HandCtrl.HitReactionPlay), AccessTools.all, null, new[] { typeof(int), typeof(bool) }, null) ?? 
			typeof(HandCtrl).GetMethod(nameof(HandCtrl.HitReactionPlay), AccessTools.all, null, new[] { typeof(HandCtrl.AibuColliderKind), typeof(bool) }, null) ;
```

Reply:
> Using int as a substitute for an enum with underlying int for GetMethod and other reflection only works in .Net 3.5 I believe.

Have you tested the original code in KKS to confirm that it doesn't work?

*******

```c#
Type.GetType("VRHandCtrl, Assembly-CSharp")?.GetMethod(nameof(HandCtrl.HitReactionPlay), AccessTools.all, null, new[] { typeof(int), typeof(bool) }, null)
```
Will this work in KKS official VR?  Or is it intended to simply return null in that case?

I'd prefer to not break compatibility with the KKS official VR module unless this plugin is confirmed to not work at all, in which case the rest of the VR related code should be removed on the KKS side.

Reply:
> Have you tested the original code in KKS to confirm that it doesn't work?

Yes, it doesn't work. Otherwise I wouldn't have changed it.

I have not tested it with the official VR module and it won't run in it anyways since the process filter is missing. It might work in official KKS VR as it is for all I know.

Even if it does not work in the KKS VR module, it doesn't make sense to increase code complexity by removing the VR code in KKS since then there's more unique code paths that can have their own bugs and need additional testing.

Reply:
Just tested the KKS VR module and the plugin does not work at all. Many classes are different from the main game as well as the KK VR module. Making it work would require changing a lot of the code into pure reflection calls.

Reply:
> Even if it does not work in the KKS VR module, it doesn't make sense to increase code complexity by removing the VR code in KKS since then there's more unique code paths that can have their own bugs and need additional testing.

Right, I forgot that you can just use the process filter to prevent it from causing issues in KKS VR.

Reply:
Yes, the plugin only has the main game process filter so it does not load in the VR module.